radix-engine-interface/src/blueprints/epoch_manager/invocations.rs

```diff
@@ -50,9 +51,62 @@ pub const EPOCH_MANAGER_NEXT_ROUND_IDENT: &str = "next_round";
 
 #[derive(Debug, Clone, Eq, PartialEq, Sbor)]
 pub struct EpochManagerNextRoundInput {
+    /// Current round number.
+    /// Please note that in case of liveness breaks, this number may be different than previous
+    /// reported `round + 1`. Such gaps are considered "round leader's fault" and are penalized
+    /// on emission, according to leader reliability statistics (see `LeaderProposalMissHistory`).
     pub round: u64,
+
+    /// A captured history of leader proposal misses since the previous reported round.
+    pub leader_proposal_misses: LeaderProposalMissHistory,
+}
+
+impl EpochManagerNextRoundInput {
+    /// Creates a "next round" input for a regular (happy-path, in terms of consensus) round
+    /// progression, i.e. no missed proposals, no fallback rounds.
+    /// Please note that the round number passed here should be an immediate successor of the
+    /// previously reported round.
+    pub fn successful(consecutive_round: u64) -> Self {
+        Self {
+            round: consecutive_round,
+            leader_proposal_misses: LeaderProposalMissHistory {
+                gap_round_leaders: Vec::new(),
+                current_round_fallback_leader: None,
+            },
+        }
+    }
 }
 
+#[derive(Debug, Clone, Eq, PartialEq, Sbor)]
+pub struct LeaderProposalMissHistory {
+    /// The validators which were leaders of the "gap" rounds (i.e. those that were not reported to
+    /// the epoch manager since the previous call; see `EpochManagerNextRoundInput::round`).
+    /// This list will contain exactly `current_call.round - previous_call.round - 1` elements; in
+    /// theory, this makes `EpochManagerNextRoundInput::round` field redundant (i.e. computable),
```

Comment:
_(note from author)_

I strongly consider removing that `round` field.
Look how it would simplify the `::successful()` constructor...
Wdyt, reviewers?

Reply:
See my comment above regarding this :)